### DISCUSSION TITLE
Is there a way to check the code injected by the java agent?

### DISCUSSION THREAD

**Discussion by SimoneGiusso** [open-telemetry/opentelemetry-java-instrumentation#9083] [Q&A]
Is there a way to check that the java agent is correctly instrumenting the application (without running it and use a visualization tool like Jager UI)

**Comment by trask** [comment]
hi @SimoneGiusso! do you mean something like this? https://github.com/open-telemetry/opentelemetry-java-examples/tree/main/telemetry-testing

**Reply by SimoneGiusso** [reply]
> hi @SimoneGiusso! do you mean something like this? https://github.com/open-telemetry/opentelemetry-java-examples/tree/main/telemetry-testing

I would be curios about the possibility to look at the code added by the agent, but I'm not sure if it possible since the code is added at runtime.

**Reply by laurit** [reply]
Add `-Dnet.bytebuddy.dump=/some/dir` to jvm arguments and run your application. Classes transformed by byte buddy should end up in the specified directory. Disassemble these classes with `javap` or whatever you like to examine what the agent added.
